integration-cli/docker_cli_build_test.go

```diff
@@ -5160,3 +5160,51 @@ func TestBuildNotVerbose(t *testing.T) {
 
 	logDone("build - not verbose")
 }
+
+func TestBuildScratchJSON(t *testing.T) {
+	defer deleteAllContainers()
+	defer deleteImages("jrun")
+
+	ctx, err := fakeContext(`FROM scratch
+ADD echo /
+RUN [ "/echo" ]
+RUN [ "/echo", "hello" ]`, map[string]string{"echo.go": `
+package main
+
+import "fmt"
+import "os"
+
+func main() {
+	fmt.Println(os.Args[1:])
+}
+`})
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer ctx.Close()
+
+	// Build 'echo' into context
+	goCmd := exec.Command("go", "build", "-o", "echo", "echo.go")
```

Comment:
Actually, what's interesting about compiling echo.go is that this might be the most portable solution since, as long as go is available, it should work on any platform.  But I guess it all depends on whether go is there or not.